src/workflow/StatedWorkflow.js

```diff
@@ -467,16 +464,35 @@ export class StatedWorkflow {
 
         const {type:clientType} = clientParams;
 
+        // testData may contain some canned data to be used without a publisher.
         if (testData !== undefined) {
             this.logger.debug(`No 'real' subscription created because testData provided for subscription params ${StatedREPL.stringify(subscriptionParams)}`);
-            const dispatcher = this.workflowDispatcher.getDispatcher(subscriptionParams);
+            const testDataAckFunctionGenerator = ((data) => {
+                return async () => {
+                    if (subscriptionParams !== undefined && Array.isArray(subscriptionParams.acks)) {
```

Comment:
do you need to throw an exception if acks is not an array. If that means something is busted, throw error

Reply:
No, we only enable acks if the array is present in subscriptionParams. Added a comment